src/Fabulous.XamarinForms/Views/Collections/CarouselView.fs

```diff
@@ -34,12 +34,18 @@ module CarouselView =
             "CarouselView_IndicatorView"
             (fun _ newValueOpt node ->
                 let handler =
-                    match node.TryGetHandler<EventHandler<IndicatorView>>(ViewRefAttributes.ViewRef.Name) with
+                    match node.TryGetHandler<Handler<IndicatorView>>(ViewRefAttributes.ViewRef.Name) with
                     | ValueSome handler -> handler
                     | ValueNone ->
                         let newHandler =
-                            EventHandler<IndicatorView>
-                                (fun _ indicatorView -> (node.Target :?> CarouselView).IndicatorView <- indicatorView)
+                            Handler<IndicatorView>
+                                (fun viewRef indicatorView ->
+                                    let carouselView = node.Target :?> CarouselView
+                                    if carouselView <> null then
+                                        carouselView.IndicatorView <- indicatorView
+                                    else
+                                        // The target has been destroyed, clean up the handler
+                                        (viewRef :?> ViewRef<IndicatorView>).ClearListeners())
```

Comment:
Might not be the best idea 🤔 
We only want to clean up the current handler, not all handlers of ViewRef because the IndicatorView might still be alive

Reply:
Would it make sense IsAlive concept in the ViewRef and then use that to clean up only the one we want ?

Reply:
The issue was with the ViewNode.Target here, not ViewRef.
Unfortunately, ViewNode.Target uses WeakRef but it doesn't warn us when the target has been disposed.
So we need to check for null and then remove the old event handler.